lib/store/src/csf/stepRunners.ts

```diff
@@ -0,0 +1,13 @@
+import { AnyFramework, StepRunner } from '@storybook/csf';
+
+export function composeStepRunners<TFramework extends AnyFramework>(
+  stepRunners: StepRunner<TFramework>[]
+): StepRunner<TFramework> {
+  return async (label, play, playContext) => {
+    const composedPlay = await stepRunners.reduce(
+      async (innerPlay, stepRunner) => async () => stepRunner(label, await innerPlay, playContext),
+      Promise.resolve(play)
+    );
```

Comment:
I thing you meant to invoke the inner play function somewhere, right? Otherwise, why the async/await? `play` is not a Promise, right? It's a function which returns a Promise. If we're not invoking it here, there's no Promise to await either, afaik. Maybe I'm just not following the logic here.

Reply:
I threw this in [a CodeSandbox](https://codesandbox.io/s/admiring-banach-3dhldp?file=/src/index.js) to figure out what's happening. It seems the logic isn't quite right because the order of invocations is wrong (I fixed that by switching to `reduceRight`), and I would also expect each runner to be able to update the context, with the next runner receiving the updated context. Arguably the `label` should be composable as well (e.g. each runner appending some text to it).

Reply:
> Maybe I'm just not following the logic here.

Reading it again I'm not quite following either, let me adjust it.

> It seems the logic isn't quite right because the order of invocations is wrong (I fixed that by switching to reduceRight),

I don't see why they need to run in any particular order, but if you feel strongly I'm happy to switch to `reduceRight`.

> I would also expect each runner to be able to update the context, with the next runner receiving the updated context. Arguably the label should be composable as well (e.g. each runner appending some text to it).

I don't see any reason why a step runner should be able to do any of that. Can you provide a use case? I'm very much of the opinion that we should start as restrictive as possible until we discover a use case that requires we make things more powerfile.

Reply:
Agreed